### PR TITLE
LPS-200110 - Create frontend unit tests to ObjectDataContainer component and Model Builder RightSidebarEmpty component

### PR DESCRIPTION
**Pull request by danvb112**
Related Issues:
https://liferay.atlassian.net/browse/LPS-200110
https://liferay.atlassian.net/browse/LPS-200109

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Nice job, @danvb112!
I have a suggestion but feel free to analyze and do the way you prefer.

We could group some tests by adding a new description, for example, in the cases that tests are related to when the object definition is published.
Another pattern I noticed in web and also in some test files on portal was that the **_it_** sentences are often on simple present.
So, we could do something like this and keep the portal pattern:

```
describe('ObjectDataContainer component', () => {
	describe('if the object definition is published', () => {
              it('shows object name input disabled', () => {...}}}
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	modifiable: false,
+	name: 'Object Name',
+	pluralLabel: {en_US: 'Plural Label Test'},
+	system: false,
+	titleObjectFieldName: '',
+} as Partial<ObjectDefinition>;
+
+describe('ObjectDataContainer component', () => {
+	it('check whether the object name input value is displayed correctly and is required', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:27]: To keep the present simple we could do something like:

```suggestion
	it('shows the object definition name input value and makes it required', () => {
		render(
```

**Review by alinedoleron — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		const objectNameInput = document.getElementById(
+			'lfr-objects__object-data-container-name'
+		);
+
+		expect(objectNameInput).toBeRequired();
+		expect(objectNameInput).toHaveValue('Object Name');
+	});
+
+	it('check when object is published whether object name input is disabled', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:50]: If you want, make a group of Published object definitions, by creating a new description before it.

**Review by alinedoleron — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+		const objectNameInput = document.getElementById(
+			'lfr-objects__object-data-container-name'
+		);
+
+		expect(objectNameInput).toBeDisabled();
+	});
+
+	it('check whether the object label input value is displayed correctly', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:72]:
```suggestion
	it('shows the object label input value', () => {
		render(
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+				setValues={() => {}}
+				values={baseValuesMock}
+			/>
+		);
+
+		expect(screen.getByDisplayValue('Label Test')).toBeInTheDocument();
+	});
+
+	it('check whether the input value of the object plural label is displayed correctly', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:90]:
```suggestion
	it('shows the object definition plural label', () => {
		render(
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+			/>
+		);
+
+		expect(
+			screen.getByDisplayValue('Plural Label Test')
+		).toBeInTheDocument();
+	});
+
+	it('check if the object table name is displayed correctly and is disabled', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:110]:
```suggestion
	it('shows object definition table name and makes it disabled', () => {
		render(
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+		);
+
+		const tableNameInput = screen.getByDisplayValue('DBTableName');
+
+		expect(tableNameInput).toBeInTheDocument();
+		expect(tableNameInput).toBeDisabled();
+	});
+
+	it('make sure active object toggle is not checked and disabled by default', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:131]: To keep the tests result easier for reading, let's make simpler and more direct sentences, like:

```suggestion
	it('shows the toggle not checked and disabled by default', () => {
		render(
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+		);
+
+		const toggleElement = screen.getByLabelText('activate-x');
+
+		expect(toggleElement).not.toBeChecked();
+		expect(toggleElement).toBeDisabled();
+	});
+
+	it('check when the object is published if the active object toggle is checked and not disabled', () => {
+		render(
```
Comment (alinedoleron) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ObjectDataContainer.spec.tsx:152]: Group this test into published object definitions.

**Comment by danvb112** [comment]
These are excellent suggestions @alinedoleron! Thank you! :smile:

**Comment by danvb112** [comment]
ci:test:sf

**Review by alinedoleron — APPROVED** [review]
Great job! LGTM! 🚀

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:forward:force

**Pull request closed without merging**